components/ws-manager/pkg/manager/internal/grpcpool/pool.go

```diff
@@ -27,13 +28,16 @@ type Pool struct {
 	factory     Factory
 	closed      bool
 	mu          sync.RWMutex
+
+	checkFn func(string) error
 }
 
 // New creates a new connection pool
-func New(factory Factory) *Pool {
+func New(factory Factory, checkFn func(string) error) *Pool {
```

Comment:
The connection validation function type could be a tad more explicit, e.g.
```
type ConnectionValidationFunc func(hostIP string) (valid bool)
```

`valid` would always indicate if we should keep the connection around. If the ConnectionValidationFunc needs to do error handling, it has to do this itself (e.g. log).
There's no sensible way we could do this outside the function right now, other than log ourselves.

Reply:
done